mmv1/products/workstations/WorkstationConfig.yaml

```diff
@@ -291,6 +298,11 @@ properties:
               values:
                 - :DELETE
                 - :RETAIN
+            - !ruby/object:Api::Type::String
+              name: 'sourceSnapshot'
+              description: |
+                Name of the snapshot to use as the source for the disk. If set, sizeGb and fsType must be empty.
```

Comment:
A ["conflicts"](https://github.com/GoogleCloudPlatform/magic-modules/blob/1d360b49c737a00ed1ad3af7f182489b388693a0/mmv1/api/type.rb#LL84C4-L84C4) tag can be added here and to the other two fields mentioned to ensure a user does not set both.

Reply:
It doesn't seem to work at the time (perhaps since it is a nested objected within a repeated field?):

```
│ Error: InternalValidate
│ 
│   with provider["registry.terraform.io/hashicorp/google-beta"],
│   on main.tf line 10, in provider "google-beta":
│   10: provider "google-beta" {
│ 
│ Internal validation of the provider failed! This is always a bug
│ with the provider itself, and not a user issue. Please report
│ this bug:
│ 
│ 1 error occurred:
│       * resource google_workstations_workstation_config: ConflictsWith: source_snapshot configuration block reference
│ (persistent_directories.0.gce_pd.0.size_gb) can only be used with TypeList and MaxItems: 1 configuration blocks
```

I have placed a `TODO` for now, but do you know any way around it?

Reply:
If that addition isn't working then it should be fine as is without the to-do even.